src/hubdata/connect_hub.py

```diff
@@ -31,12 +43,63 @@ def __init__(self, hub_dir: Path):
         :param hub_dir: Path to a hub's root directory. see: https://docs.hubverse.io/en/latest/user-guide/hub-structure.html
         """
 
+        # set hub_dir, first checking for existence
         self.hub_dir = hub_dir
         if not self.hub_dir.exists():
             raise RuntimeError(f'hub_dir not found: {self.hub_dir}')
 
-        with open(self.hub_dir / 'hub-config' / 'tasks.json') as fp:
+        # set self.admin, first checking for admin.json existence
+        admin_json_file = self.hub_dir / 'hub-config' / 'admin.json'
+        if not admin_json_file.exists():
+            raise RuntimeError(f'admin.json not found: {admin_json_file}')
+
+        with open(admin_json_file) as fp:
+            self.admin: dict = json.load(fp)
+
+        # set self.tasks, first checking for tasks.json existence
+        tasks_json_file = self.hub_dir / 'hub-config' / 'tasks.json'
+        if not tasks_json_file.exists():
+            raise RuntimeError(f'tasks.json not found: {tasks_json_file}')
+
+        with open(tasks_json_file) as fp:
             self.tasks: dict = json.load(fp)
 
-        with open(self.hub_dir / 'hub-config' / 'model-metadata-schema.json') as fp:
+        # set schema
+        self.schema = create_hub_schema(self.tasks)
+
+        # set self.model_metadata_schema, first checking for model-metadata-schema.json existence
+        model_metadata_schema_file = self.hub_dir / 'hub-config' / 'model-metadata-schema.json'
+        if not model_metadata_schema_file.exists():
+            raise RuntimeError(f'model-metadata-schema.json not found: {model_metadata_schema_file}')
+
+        with open(model_metadata_schema_file) as fp:
```

Comment:
Do we want it to be a blocker? I mean do we want to have an error if the metadata schema is missing or just a warning but still have the function continuing?

Reply:
I don't think its a blocker in the R version

Reply:
Thank you. Do you suggest a warning instead of an error, or silent?

Reply:
[lucie replied in slack] I think "silent" for now, to have the same behavior as the R package and we can open an issue later if it's something that we need to change?